parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -36,18 +36,36 @@ configureTransparentHugePage() {
 
 {{- if ShouldConfigSwapFile}}
 configureSwapFile() {
-    SWAP_SIZE_KB=$(expr {{GetSwapFileSizeMB}} \* 1000)
-    DISK_FREE_KB=$(df /dev/sdb1 | sed 1d | awk '{print $4}')
-    if [[ ${DISK_FREE_KB} -gt ${SWAP_SIZE_KB} ]]; then
-        SWAP_LOCATION=/mnt/swapfile
-        retrycmd_if_failure 24 5 25 fallocate -l ${SWAP_SIZE_KB}K ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        chmod 600 ${SWAP_LOCATION}
-        retrycmd_if_failure 24 5 25 mkswap ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        retrycmd_if_failure 24 5 25 swapon ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        retrycmd_if_failure 24 5 25 swapon --show | grep ${SWAP_LOCATION} || exit $ERR_SWAP_CREAT_FAIL
-        echo "${SWAP_LOCATION} none swap sw 0 0" >> /etc/fstab
+    # https://learn.microsoft.com/en-us/troubleshoot/azure/virtual-machines/troubleshoot-device-names-problems#identify-disk-luns
+    swap_size_kb=$(expr {{GetSwapFileSizeMB}} \* 1000)
+    symlinks=$(ls -la /dev/disk/azure)
+
+    if [[ -n "$(echo "${symlinks}" | grep "resource-part1")" ]]; then
+        echo "Will use resource disk for swap file"
+        resource_disk_name=$(echo "${symlinks}" | grep "resource-part1" | awk '{print $11}')
+    elif [[ -n "$(echo "${symlinks}" | grep "root-part1" | aws '{print $11}')" ]]; then
+        echo "Resource disk unavailable, will use OS disk for swap file"
+        resource_disk_name=$(echo "${symlinks}" | grep "root-part1" | awk '{print $11}')
     else
-        echo "Insufficient disk space creating swap file: request ${SWAP_SIZE_KB} free ${DISK_FREE_KB}"
+        echo "Unable to identify a disk for swap file storage, both 'root-part1' and 'resource-part1' disk labels are absent"
```

Comment:
root-part1 being absent should be irrelevant, just pick a predefined dir (os disk has to exist)